dev/Repeater/FlexboxLayout.cpp

```diff
@@ -0,0 +1,468 @@
+{
+    m_justifyContent = value;
+}
+
+winrt::FlexboxAlignItems FlexboxLayout::AlignItems()
+{
+    return m_alignItems;
+}
+
+void FlexboxLayout::AlignItems(winrt::FlexboxAlignItems const& value)
+{
+    m_alignItems = value;
+}
+
+winrt::FlexboxAlignContent FlexboxLayout::AlignContent()
+{
+    return m_alignContent;
+}
+
+void FlexboxLayout::AlignContent(winrt::FlexboxAlignContent const& value)
+{
+    m_alignContent = value;
+}
+
+bool FlexboxLayout::IsHorizontal()
+{
+    return (m_direction == winrt::FlexboxDirection::Row || m_direction == winrt::FlexboxDirection::RowReverse);
+}
+
+bool FlexboxLayout::IsReversed()
+{
+    return (m_direction == winrt::FlexboxDirection::RowReverse || m_direction == winrt::FlexboxDirection::ColumnReverse);
+}
+
+bool FlexboxLayout::IsWrapping()
+{
+    return (m_wrap != winrt::FlexboxWrap::NoWrap);
+}
+
+float FlexboxLayout::MainAxis(winrt::Size const& value)
+{
+    return (IsHorizontal() ? value.Width : value.Height);
+}
+
+float FlexboxLayout::CrossAxis(winrt::Size const& value)
+{
+    return (IsHorizontal() ? value.Height : value.Width);
+}
+
+
+winrt::Size FlexboxLayout::CreateSize(float mainAxis, float crossAxis)
+{
+    return IsHorizontal() ?
+        winrt::Size{ mainAxis, crossAxis } :
+        winrt::Size{ crossAxis, mainAxis };
+}
+
+winrt::Point FlexboxLayout::CreatePoint(float mainAxis, float crossAxis)
+{
+    return IsHorizontal() ?
+        winrt::Point{ mainAxis, crossAxis } :
+        winrt::Point{ crossAxis, mainAxis };
+}
+
+std::vector<winrt::UIElement> FlexboxLayout::ChildrenSortedByOrder(winrt::NonVirtualizingLayoutContext const& context)
+{
+    std::vector<winrt::UIElement> sorted;
+
+   
+    for (auto const& child : context.Children())
+    {
+        sorted.push_back(child);
+    }
+    std::sort(sorted.begin(), sorted.end(), [](winrt::UIElement const& a, winrt::UIElement const& b)
+        {
+            return (GetOrder(a) < GetOrder(b));
+        });
+
+    return sorted;
+}
+
+void FlexboxLayout::InitializeForContextCore(winrt::LayoutContext const& context)
+{
+    auto state = context.LayoutState();
+    winrt::com_ptr<FlexboxLayoutState> flexboxState = nullptr;
+    if (state)
+    {
+        flexboxState = state.as<FlexboxLayoutState>();
+    }
+
+    if (!flexboxState)
+    {
+        if (state)
+        {
+            throw winrt::hresult_error(E_FAIL, L"LayoutState must derive from FlexboxLayoutState.");
+        }
+
+        flexboxState = winrt::make_self<FlexboxLayoutState>();
+    }
+
+    context.LayoutStateCore(*flexboxState);
+}
+
+void FlexboxLayout::UninitializeForContextCore(winrt::LayoutContext const& context)
+{
+}
+
+winrt::Size FlexboxLayout::MeasureOverride(
+    winrt::NonVirtualizingLayoutContext const& context,
+    winrt::Size const& availableSize)
+{
+    auto state = context.LayoutState().as<FlexboxLayoutState>();
+    state->Rows.clear();
+
+    unsigned int itemsInRow = 0;
+    float growInRow = 0.0;
+    float shrinkInRow = 0.0;
```

Comment:
this and others are float, but the properties they get set from are double. Use double where the value is double :)

Reply:
Oops, guess that's why I had to cast my variables in another spot